internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -149,47 +149,30 @@ func (r *GitJobReconciler) createJob(ctx context.Context, gitRepo *v1alpha1.GitR
 	if err := controllerutil.SetControllerReference(gitRepo, job, r.Scheme); err != nil {
 		return err
 	}
-	err = r.Create(ctx, job)
-	if err != nil {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		var gitRepoFromCluster v1alpha1.GitRepo
-		err := r.Get(
-			ctx,
-			types.NamespacedName{Name: gitRepo.Name, Namespace: gitRepo.Namespace},
-			&gitRepoFromCluster,
-		)
-		if err != nil {
-			return err
-		}
-		gitRepoFromCluster.Status.ObservedGeneration = gitRepoFromCluster.Generation
-
-		return r.Status().Update(ctx, &gitRepoFromCluster)
-	})
-
+	return r.Create(ctx, job)
 }
 
 func (r *GitJobReconciler) updateStatus(ctx context.Context, gitRepo *v1alpha1.GitRepo, job *batchv1.Job) error {
+	// Convert the Job to unstructured format
 	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(job)
 	if err != nil {
 		return err
 	}
 	uJob := &unstructured.Unstructured{Object: obj}
+
+	// Compute the status
 	result, err := status.Compute(uJob)
 	if err != nil {
 		return err
 	}
 
-	gitRepo.Status.GitJobStatus = result.Status.String()
-	for _, con := range result.Conditions {
-		condition.Cond(con.Type.String()).SetStatus(gitRepo, string(con.Status))
-		condition.Cond(con.Type.String()).SetMessageIfBlank(gitRepo, con.Message)
-		condition.Cond(con.Type.String()).Reason(gitRepo, con.Reason)
-	}
+	// Prepare the computed values for updating
+	gitJobStatus := result.Status.String() // General job status
```

Comment:
I don't think we need this variable anymore 🤷 
And the next line too

Reply:
Of course, I removed them